### ISSUE TITLE
3.6.0 docusarus won't compile with the new docusarus faster plugin

### ISSUE THREAD

**Issue by Celtech** [PaloAltoNetworks/docusaurus-openapi-docs#1014]
## Describe the bug

If you follow the 3.6.0 guide detailed here: https://docusaurus.io/blog/releases/3.6#docusaurus-faster
and install / set up the faster plugin you will be met with the following error when trying to compile:

<img width="1370" alt="image" src="https://github.com/user-attachments/assets/cfc96d3d-7185-4746-99dd-70b8508aab41">

## Steps to reproduce

1.
2.
3.
4.

## Your Environment

- Version used:
- Environment name and version (e.g. Chrome 59, node.js 5.4, python 3.7.3):
- Operating System and version (desktop or mobile):
- Link to your project:

**Comment by pmarschik** [comment]
I suspect the instantiation of `NodePolyfillPlugin` in `packages/docusaurus-theme-openapi-docs/src/index.ts` is the culprit here.

With the following faster configuration my project builds (i.e. disable `rspackBundler`):

```typescript
  future: {
    experimental_faster: {
      swcJsLoader: true,
      swcJsMinimizer: true,
      swcHtmlMinimizer: true,
      lightningCssMinimizer: true,
      rspackBundler: false,
      mdxCrossCompilerCache: true,
    },
  },
```

But ideally we should be able to ose the new rspack bundler.

**Comment by alvinometric** [comment]
That should be fixed with https://github.com/PaloAltoNetworks/docusaurus-openapi-docs/pull/1003

It'a merged but not released yet

**Comment by slorber** [comment]
Yes support is merged but you'll have to use the next release of this project to enable Rspack, or force resolution to v3 of this dependency: https://github.com/Richienb/node-polyfill-webpack-plugin/releases/tag/v3.0.0

Unfortunately, I don't really have time to maintain this extra repository and I'm not confident with the codebase enough to merge/release things there on behalf of the original authors 😅

**Comment by sserrata** [comment]
We're always happy to have you @slorber 😅...

Just a reminder that we publish canary releases with each merge to main, in the event you need new features sooner. You can discover the latest canary release version in the site/README badges or by visiting our package(s) on npmjs.org.  

In general, we try to squeeze in at least a few changes/fixes into each release, hence why we didn't publish immediately. We also can't control for breaking changes introduced with new Docusaurus releases, e.g. changes requiring refactors or upgrades to dependencies, etc., which can further delay support.

**Comment by dogun-anduril** [comment]
The canary publish looks to have failed though? https://github.com/PaloAltoNetworks/docusaurus-openapi-docs/actions/runs/11669994426/job/32493230188

**Comment by KashifSaadat** [comment]
Hi @dogun-anduril, this has been fixed now in the `main` branch. I have tested with canary release `0.0.0-949` and can confirm it's working successfully with the docusaurus-faster settings (and `rspackBundler: true`). My build time has reduced from 11 minutes to ~3 minutes!

Thank you very much @slorber & @sserrata ! 🎉

**Comment by slorber** [comment]
Awesome 🥳  thanks

**Comment by slorber** [comment]
Released in v4.3: https://github.com/PaloAltoNetworks/docusaurus-openapi-docs/discussions/1043
